fake_xml_http_request.js

```diff
@@ -272,7 +272,7 @@
       verifyState(this);
 
       if (!/^(get|head)$/i.test(this.method)) {
-        if (!this.requestHeaders["Content-Type"] && !(data || '').toString().match('FormData')) {
+        if (!this.requestHeaders["Content-Type"] && !this.requestHeaders["content-type"] && !(data || '').toString().match('FormData')) {
```

Comment:
I wonder if instead you should be doing something like:
```
const hasContentTypeHeader = Object.keys(this.requestHeaders).some((key) => key.toLowerCase() === 'content-type')

if (!hasContentTypeHeader && !(data || '').toString.match('FormData')) {
```
Granted I'm guessing we'd need to use ES5 over ES6.